src/middlewared/middlewared/plugins/usage.py

```diff
@@ -12,6 +12,11 @@ class Config:
         private = True
 
     async def start(self):
+        if not (
+            await self.middleware.call('system.general.config')
+        )['usage_collection']:
+            return False
```

Comment:
You only need to skip gather & send section, not the entire block, otherwise if data collection is re-enabled, loop won't be resumed.